libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.m

```diff
@@ -830,11 +833,18 @@ - (void)dismissSnapshot
             self.snapshotDismissalAction(_snapshotViewController);
             if ([SFSecurityLockout isPasscodeNeeded]) {
                 [SFSecurityLockout validateTimer];
+            } else{
+                [SFSecurityLockout validateTimer];
+                [lastActiveWindow presentWindow];
             }
         } else {
-            [[SFSDKWindowManager sharedManager].snapshotWindow dismissWindowAnimated:NO withCompletion:^{
+            __block SFSDKWindowContainer *activeWindow = lastActiveWindow;
```

Comment:
Snapshot window needs special handling. We could be coming back to a main window or a passcode window. lastActiveWindow is captured before snapshot is presented.